src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,92 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Node[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_CAPACITY];
+        threshold = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node currentNode = table[getIndex((key))];
+        Node<K, V> newNode = new Node<>(Objects.hashCode(key), key, value, null);
+        if (currentNode == null) {
+            table[getIndex((key))] = newNode;
+            size++;
+        }
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                currentNode.value = value;
+                break;
+            } else if (currentNode.next == null) {
+                currentNode.next = newNode;
+                size++;
+            }
+            currentNode = currentNode.next;
+        }
+        if (size >= threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node currentNode = table[getIndex(key)];
+        while (currentNode != null) {
+            if (Objects.equals(key, currentNode.key)) {
+                return (V) currentNode.value;
```

Comment:
do we need to cast currentNode.value to (V)?

Reply:
Agree